staging/src/k8s.io/component-base/cli/flag/flags_test.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flag
+
+import (
+	"bytes"
+	"github.com/go-logr/logr"
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/klog/v2"
+	"testing"
+)
+
+type FakeLogger struct {
```

Comment:
I looked (could have missed something) and was surprised that this doesn't exist as a reusable thing for tests.

Might be a good PR to add to klog?